### ISSUE TITLE
Inconsistent kallsyms data

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1280]
cc @arndb 

working around #1195 and #1271, with LLVM_IAS=1 allmodconfig

```
$ ARCH=arm CROSS_COMPILE=arm-linux-gnueabi- make LLVM=1 LLVM_IAS=1 -j72 allmodconfig
<apply workarounds>
$ ARCH=arm CROSS_COMPILE=arm-linux-gnueabi- make LLVM=1 LLVM_IAS=1 -j72
...
  LD      vmlinux
  SORTTAB vmlinux
  SYSMAP  System.map
Inconsistent kallsyms data
Try make KALLSYMS_EXTRA_PASS=1 as a workaround
```
so CONFIG_KALLSYMS which gets selected by a lot of things looks like it's broken with LLVM_IAS=1.

**Comment by arndb** [comment]
I played around it some more, and found that this is mainly a function of the image size -- whenever .text grows beyond 32MB (for arm) or 8MB (for thumb2), there is a good chance to hit the bug, though changing random Kconfig options can also make it go away.

The other thing I found is that it eventually does converge if you add further iterations to kallsyms. The problem is that something (either lld or ias) adds symbols such as "__ThumbV7PILongThunk_unregister_chrdev_region" when it adds a veneer. These get added to the kallsyms symbol table, which enlarges the kernel image and leads to further veneers getting added in the next iteration.

Preparing a patch now to add a few more prefixes to the kallsyms blocklist.

**Comment by nickdesaulniers** [comment]
cc @groeck

**Comment by arndb** [comment]
Patch applied for 5.12: https://lore.kernel.org/lkml/CAK7LNAQPKaL1xbFfdWYSPZ8S4cfyZ2N2BriVPa-OE5A6c5d_9A@mail.gmail.com/

**Comment by dileks** [comment]
Patch in `kbuild.git#fixes`.

[1] https://git.kernel.org/pub/scm/linux/kernel/git/masahiroy/linux-kbuild.git/commit/?h=fixes&id=efe6e3068067212b85c2d0474b5ee3b2d0c7adab
[2] https://git.kernel.org/masahiroy/linux-kbuild/c/efe6e3068067212b85c2d0474b5ee3b2d0c7adab

Update: Add shorter link in [2] (points to the same commit)

**Comment by nathanchance** [comment]
Merged into mainline: https://git.kernel.org/linus/efe6e3068067212b85c2d0474b5ee3b2d0c7adab
